File: ios/bridges/SoundPlayer/SoundPlayer.m
//
//  SoundPlayer.m
//  rn_arch_demo
//
//  Created by Rafael Ruiz on 21/01/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

#import "React/RCTBridgeModule.h"

@interface
RCT_EXTERN_REMAP_MODULE(SoundPlayerIOS, SoundPlayer, NSObject)

RCT_EXTERN_METHOD(play)
RCT_EXTERN_METHOD(stop)
@end

File: ios/bridges/SoundPlayer/SoundPlayer.swift
//
//  SoundPlayer.swift
//  rn_arch_demo
//
//  Created by Rafael Ruiz on 21/01/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation
import SwiftySound

@objc(SoundPlayer) class SoundPlayer: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func play() {
    Sound.play(file: "morado.mp3")
  }
  
  @objc func stop() {
    Sound.stopAll()
  }
}
